### PR TITLE
Fixes to get matlab wrapper working properly

### PR DESCRIPTION
**Pull request by varunagrawal**
Just the title. Recent churn and lack of Matlab CI allowed some bugs to creep in so just solving those bugs.

### CODE REVIEW

**Review by dellaert — APPROVED** [review]
Thanks!!! I guess these came to light when testing MATLAB?

**Comment by varunagrawal** [comment]
Yes

**Comment by dellaert** [comment]
@mattking-smith FYI

File: python/gtsam/preamble/inference.h [hunk 1]
```diff
@@ -10,6 +10,3 @@
  * Without this they will be automatically converted to a Python object, and all
  * mutations on Python side will not be reflected on C++.
  */
-
-#include <pybind11/stl.h>
```
Comment (dellaert) [inline python/gtsam/preamble/inference.h:14]: @varunagrawal : why was this removed? I think this broke DotWriter
